### DISCUSSION TITLE
Cannot fail jobs: Error: Missing lock for job 1505. failed

### DISCUSSION THREAD

**Discussion by AlessioGr** [taskforcesh/bullmq#1894] [Q&A]
```ts
  const creatorWorker = new Worker<CreatorInputData, CreatorOutputData>(
    queueNameCreator,
    async (job: Job, token: string) => {
      console.log(
        Date.now() + ' - Worker is processing an embeddings creator job...',
      );
...
      try {
        ...
      } catch (e) {
        if(e.response.status === 422) {
          // Unprocessable entity
          console.log(' 422 error, removing job');
          await job.moveToFailed(e, token);
          throw new UnrecoverableError(e.response.data.detail);
          return;
        } else {
          throw e;
        }
      }
      return toReturn;
    },
    {
      autorun: true,
      connection: redisConnection,
      limiter: {
        max: 4
        duration: 1000 * 6,
      },
      concurrency: 4,
      removeOnComplete: {
        count: 2,
      },
      removeOnFail: {
        count: 10,
      },
      maxStalledCount: 10000000,
    },
  );
  creatorWorker.on('failed', (job: Job, error: Error) => {
    console.log('Embeddings creator job failed. Error', error);
  });
  creatorWorker.on(
    'completed',
    async (job: Job, returnvalue: CreatorOutputData) => {
     ...
    },
  );
```
When the error is caught, it does get delayed, but throws the following error:

1683996065811 - Worker is processing an embeddings creator job...
 422 error, removing job
Error: Missing lock for job 1431. failed
    at Scripts.finishedErrors (/Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/bullmq/src/classes/scripts.ts:332:16)
    at Job.moveToFailed (/Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/bullmq/src/classes/job.ts:648:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async handleFailed (/Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/bullmq/src/classes/worker.ts:658:11)
    at async Worker.retryIfFailed (/Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/bullmq/src/classes/worker.ts:874:16)
    
    
Why is that? I'm passing the token in the moveToFailed function, so why am I getting a "Missing lock" error?

**Comment by manast** [comment]
Is that exact code running on the latest version of BullMQ producing that error? I am wondering because we have a test cases that cover exact this case: https://github.com/taskforcesh/bullmq/blob/master/tests/test_worker.ts#L2306

**Reply by AlessioGr** [reply]
@manast I'm sorry I provided the wrong worker in the code. I have updated my question and the code - the problem lies when failing jobs using moveToFailed, not when delaying jobs.

Also: here's the full file: https://gist.github.com/AlessioGr/085e0553d830d3ae04ec20cb63b816f2
though the question here should contain the relevant parts I've extracted

**Comment by AlessioGr** [comment]
Closing this in favor of the issue #1896 where I managed to reproduce the issue
